api/google.go

```diff
@@ -63,3 +72,69 @@ func GetSCCHealthStatus(url string) int {
 	}
 	return 1
 }
+
+// GetSCCSourcesByName returns Security Command Center sources for given numeric orgID,
+// filtered by name by given regex
+// original: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/securitycenter/findings/list_sources.go
+func GetSCCSourcesByName(orgID string, nameRegex string) (map[string]string, error) {
+	regex, err := regexp.Compile(nameRegex)
+	if err != nil {
+		return nil, errors.Wrap(err, "error compiling nameRegex")
+	}
+	// Instantiate a context and a security service client to make API calls.
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "securitycenter.NewClient")
+	}
+	defer client.Close() // Closing the client safely cleans up background resources.
+
+	req := &securitycenterpb.ListSourcesRequest{
+		Parent: fmt.Sprintf("organizations/%s", orgID),
+	}
+	it := client.ListSources(ctx, req)
+	result := map[string]string{}
+	for {
+		source, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "it.Next")
+		}
+
+		if match := regex.MatchString(source.DisplayName); match {
+			result[source.Name] = source.DisplayName
+		}
+	}
+	return result, nil
+}
+
+// GetSCCLatestEventTime return map of sources and their latest event update time difference with now
+// original: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/securitycenter/findings/list_filtered_findings.go
+func GetSCCLatestEventTime(sources map[string]string) (map[string]time.Duration, error) {
+	result := make(map[string]time.Duration)
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "securitycenter.NewClient")
+	}
+	defer client.Close() // Closing the client safely cleans up background resources.
+
+	// process just one event with newest update date for every given source
+	for id, name := range sources {
+		req := &securitycenterpb.ListFindingsRequest{
+			Parent:   id,
+			OrderBy:  `event_time desc`,
+			PageSize: 1,
+		}
+		it := client.ListFindings(ctx, req)
```

Comment:
The semantics of this call tells there might be more than 1 result, which doesn't make sense from the business logic perspective: it is getting an iterator where contextually might be 0, 1, 2 and more elements. A more explicit single-unit container is more preferable here.